### ISSUE TITLE
Plans?

### ISSUE THREAD

**Issue by danbri** [pfps/shacl#1]
Hi Peter! 

What are you plans here? I was looking around to see if I could find a modern/compliant SHACL implementation in Python. Do you think you'll develop this codebase further?

**Comment by pfps** [comment]
I hadn't planned on doing anything further to the codebase.


The code may not be completely complaint with the current definition.  I wrote
it to show how the definition of SHACL core could be improved.  However, this
code should be compliant to most of the definition of SHACL core.


peter

**Comment by danbri** [comment]
Thanks, Peter. I couldn't find much else for SHACL, but for ShEx at least there is now https://github.com/hsolbrig/PyShEx

**Comment by pfps** [comment]
Yeah, I'm quite surprised given the errors in http://shex.io/shex-semantics/ and the ill-founded nature of the semantics it defines.

**Comment by nicholascar** [comment]
I’m keen to devote effort to ensuring there is up-to-spec Python tooling for SHACL. 

We are using SHACL in a number of places (a validation, UI generation, profile definitions) in a number of projects and have about 5 of us at CSIRO keen to do this all in Python.

Should I fork and PR back or just start something new?

Nick

**Comment by VladimirAlexiev** [comment]
@nicholascar do you have some examples of UI generation?

**Comment by pfps** [comment]
As I said, I'm not planning on doing anything with this.

I expect that if you wan this to conform to SHACL then you should fork and not PR back.

**Comment by Fak3** [comment]
@nicholascar do you have any progress on the updated version?

@pfps What is the license of this code? I would like to port it to python3 and make source code available.

**Comment by pfps** [comment]
It's a contribution to a W3C working group, so whatever that license is.


peter

**Comment by nicholascar** [comment]
@VladimirAlexiev (I recognise you from the LODE Issues) and @Fak3:

I’m still looking for other tooling but will start work in mid-July if I can’t find anything for projects starting then. Will likely follow the JavaScript SHACL Playground Implementation.

I will put together a few notes about the UI generation and post here: my student is almost finished.

**Comment by VladimirAlexiev** [comment]
@nicholascar Great! When done, please register the implementation here: https://github.com/validatingrdf/validatingrdf.github.io/issues/14

**Comment by nicholascar** [comment]
Here we go, a pure Python implementation of a SHACL validator: https://github.com/RDFLib/pySHACL

Core constraints done now and SPARQL-based constraints will be completed shortly.
